core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -189,8 +185,7 @@ private Translog openTranslog(EngineConfig engineConfig, IndexWriter writer, boo
             }
             translogConfig.setTranslogGeneration(generation);
             if (generation != null && generation.translogUUID == null) {
-                // only upgrade on pre-2.0 indices...
-                Translog.upgradeLegacyTranslog(logger, translogConfig);
+                throw new IndexFormatTooOldException("trasnlog", "translog has no generation nor a UUID - this might be an index from a previous version consider upgrading to N-1 first");
```

Comment:
feels like this can better go into loadTranslogIdFromCommit() . We do more checks that maybe can go away...  (also if we get here we know that generation is non-null.